pkg/virt-launcher/notify-client/client.go

```diff
@@ -146,25 +147,46 @@ func libvirtEventCallback(c cli.Connection, domain *api.Domain, libvirtEvent lib
 				client.SendDomainEvent(event)
 				events <- event
 			}
+		} else if interfaceStatus != nil {
+			domain.Status.Interfaces = *interfaceStatus
+			event := watch.Event{Type: watch.Modified, Object: domain}
+			client.SendDomainEvent(event)
+			events <- event
 		}
 		client.SendDomainEvent(watch.Event{Type: watch.Modified, Object: domain})
 	}
 }
 
-func (c *NotifyClient) StartDomainNotifier(domainConn cli.Connection, deleteNotificationSent chan watch.Event, vmiUID types.UID) error {
-
+func (c *NotifyClient) StartDomainNotifier(domainConn cli.Connection, deleteNotificationSent chan watch.Event, vmiUID types.UID, qemuAgentPollerInterval *time.Duration) error {
 	eventChan := make(chan libvirtEvent, 10)
+	agentUpdateChan := make(chan agentpoller.AgentUpdateEvent, 10)
+
 	reconnectChan := make(chan bool, 10)
 
 	domainConn.SetReconnectChan(reconnectChan)
 
+	agentPoller := agentpoller.CreatePoller(domainConn, vmiUID, agentUpdateChan, qemuAgentPollerInterval)
+
 	// Run the event process logic in a separate go-routine to not block libvirt
 	go func() {
 		for {
 			select {
 			case event := <-eventChan:
-				libvirtEventCallback(domainConn, util.NewDomainFromName(event.Domain, vmiUID), event, c, deleteNotificationSent)
+				domain := util.NewDomainFromName(event.Domain, vmiUID)
+				eventCallback(domainConn, domain, event, c, deleteNotificationSent, nil)
+				agentPoller.UpdateDomain(domain)
+				if event.AgentEvent != nil {
+					if event.AgentEvent.State == libvirt.CONNECT_DOMAIN_EVENT_AGENT_LIFECYCLE_STATE_CONNECTED {
+						agentPoller.Start()
+					} else if event.AgentEvent.State == libvirt.CONNECT_DOMAIN_EVENT_AGENT_LIFECYCLE_STATE_DISCONNECTED {
+						agentPoller.Stop()
+					}
+				}
 				log.Log.Info("processed event")
+			case agentUpdate := <-agentUpdateChan:
```

Comment:
I think that currently you would flip the interface status. Just create `interfaceStatuses` outside of the for-loop, just update it when a new agentUpdate comes in, and always call the callback with `interfaceStatuses`.

Reply:
Done